contents/tutorials/anthropic-analytics.md

```diff
@@ -0,0 +1,357 @@
+  title: "Create Next App",
+  description: "Generated by create next app",
+};
+
+export default function RootLayout({ children }) {
+  return (
+    <html lang="en">
+      <PHProvider>
+        <body className={inter.className}>{children}</body>
+      </PHProvider>
+    </html>
+  );
+}
+```
+
+Then replace the code in `page.js` with our basic layout and functionality. You can find your Anthropic API key [here](https://console.anthropic.com/settings/keys).
+
+```js file=app/page.js
+'use client'
+
+import { useState } from 'react';
+import { usePostHog } from 'posthog-js/react'
+import Anthropic from '@anthropic-ai/sdk';
+
+const models = [
+  {
+    name: 'claude-3-opus-20240229',
+    token_input_cost: 0.00001500,
+    token_output_cost: 0.00007500
+  },
+  {
+    name: 'claude-3-sonnet-20240229',
+    token_input_cost: 0.00000300,
+    token_output_cost:  0.00001500
+  },
+    {
+    name: 'claude-3-haiku-20240307',
+    token_input_cost: 0.00000025,
+    token_output_cost: 0.00000125
+  },
+]
+
+export default function Home() {
+  const [userInput, setUserInput] = useState('');
+  const [response, setResponse] = useState('');
+  const [selectedModel, setSelectedModel] = useState(models[0]);
+  const posthog = usePostHog()
+
+  const fetchResponse = async () => {
+    try {
+      const anthropic = new Anthropic({
+        apiKey: '<your_anthropic_api_key>', 
+        baseURL: window.location.origin + '/anthropic/',
+      });  
+      
+      setResponse('Generating...');
+      const msg = await anthropic.messages.create({
+        model: selectedModel.name,
+        max_tokens: 4096,
+        messages: [{ role: "user", content: userInput }],
+      });
+
+      const response = msg.content[0].text
+      setResponse(response);
+    } catch (error) {
+      setResponse(error.message);
+    }
+  };
+
+  const handleInputChange = (event) => {
+    setUserInput(event.target.value);
+  };
+
+  const handleModelChange = (event) => {
+    setSelectedModel(models.filter(m => (m.name === event.target.value))[0]);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    fetchResponse();
+  };
+
+  return (
+    <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', justifyContent: 'center', minHeight: '100vh', gap: '20px' }}>
+      <form onSubmit={handleSubmit}>
+        <input
+          type="text"
+          value={userInput}
+          onChange={handleInputChange}
+          placeholder="Type your message"
+        />
+        <button type="submit">Send</button>
+      </form>
+      <select value={selectedModel.name} onChange={handleModelChange}>
+        {models.map((model, index) => (
+          <option key={index} value={model.name}>
+            {model.name}
+          </option>
+        ))}
+      </select>     
+      <label>API Response:</label>
+      <label>{response}</label>
+    </div>
+  );
+};
+```
+
+Lastly, replace the code in `next.config.mjs` with the following:
```

Comment:
Rewrites are relatively inconsistent in their functionality (I've learned this from doing reverse proxies). Why can't we just use the API endpoint directly?

Reply:
Was the suggested solution re the cors issue -https://www.reddit.com/r/ClaudeAI/comments/1b9pn20/cors_error_in_anthropic_api_while_calling_from/